services/api.ts

```diff
@@ -48,3 +60,37 @@ export const AddToReadingList = async (bookId: number) => {
 
   return response
 }
+
+export const MarkAsUnread = async (bookId: number) => {
+  const payload = {
+    id: bookId
+  }
+  const url = `${API_ENDPOINT}/v1/books/mark_as_unread`
+
+  const response = await fetchApi(
+    url,
+    'POST',
+    {
+      variables: JSON.stringify(payload),
+    }
```

Comment:
I'm curious on this API endpoint. Why are you passing the payload as a JSON string? Can the backend properly decode it? 🤔 

What I'd suggest instead:

```
  const response = await fetchApi(
    url,
    'POST',
    payload,
```

Then your `fetchApi` in https://github.com/karlxavier/bookshelf-client/blob/a8d4f77052b2374217cbed8d4268ec3c8e49da9b/services/api.ts#L4 can be changed to:

```

const fetchApi = async(url: string, method: string, payload: any) => {
    const options = {
    method: method,
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${accessToken}`
    },
    body: payload
  }

}
```

I know the `any` here goes against my comment before of not allowing `any type` - but for this scenario its fine. 

All your API functions have properly typed params. Then, the API function just calls this function to generate the request. This function shouldn't care about what the `payload` type is, only that it should accept it because its caller has already properly typed it. 

Please let me know if the above is confusing - we can talk about this in pairing later